src/ompl/base/PlannerTerminationCondition.h

```diff
@@ -120,6 +121,32 @@ namespace ompl
 
         /** \brief Return a termination condition that will become true \e duration seconds in the future (wall-time), but is checked in a separate thread, every \e interval seconds; \e interval must be less than \e duration */
         PlannerTerminationCondition timedPlannerTerminationCondition(double duration, double interval);
+
+        /** \brief Return a termination condition that will become true as soon as the problem definition has an exact solution */
+        PlannerTerminationCondition exactSolnPlannerTerminationCondition(ompl::base::ProblemDefinitionPtr pdef);
+
+        /** \brief A class to run a planner for a specific number of iterations. Casts to a PTC for use with Planner::solve */
+        class IterationTerminationCondition
```

Comment:
Maybe this needs a better name? It's not truly number of iterations, it's the number of calls to the PTC, which, depending on how a specific planner uses it, may not map 1:1 to iterations.

I used this for inspecting how a planner was behaving and to make videos of the search process. You're absolutely right that it would not be useful in comparing planners across iterations, but I do think it's useful to be able to look at a planner iteration by iteration...

I don't remember why I implemented it like this. To be honest, it may have been the only way I could figure out how to do it. :worried: